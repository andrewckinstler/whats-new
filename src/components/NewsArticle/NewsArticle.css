img {
  width: 100%;
}

article {
  background-color: #FFF;
}

p, h3, a {
  color: #000;
}