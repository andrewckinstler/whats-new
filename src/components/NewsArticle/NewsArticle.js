import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({headline, img, description, url}) => {
  return (
    <article>
      <div className='image' style={{backgroundImage: `url(${img})`}}>
      </div>
      <div className='text'>
      <h3>{headline}</h3>
      <p>{description}</p>
      </div>
      <div className='link'>
      <a href={url}>Link to article</a>
      </div>
    </article>
  )
}

export default NewsArticle;