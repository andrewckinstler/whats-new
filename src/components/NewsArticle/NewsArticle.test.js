import React from 'react';
import ReactDOM from 'react-dom';
import NewsArticle from './NewsArticle';
import { shallow } from 'enzyme';


it('should correctly match the snapshot', () => {
  const wrapper = shallow(<NewsArticle 
    headline='hello world'
    img='image'
    description='this is a description'
    url='www.url.com'
  />)

  expect(wrapper).toMatchSnapshot();
})

