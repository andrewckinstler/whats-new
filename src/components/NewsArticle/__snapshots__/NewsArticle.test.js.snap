// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly match the snapshot 1`] = `
<article>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(image)",
      }
    }
  />
  <div
    className="text"
  >
    <h3>
      hello world
    </h3>
    <p>
      this is a description
    </p>
  </div>
  <div
    className="link"
  >
    <a
      href="www.url.com"
    >
      Link to article
    </a>
  </div>
</article>
`;
